src/main/java/org/sonarlint/intellij/server/RequestProcessor.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015-2020 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.server
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.intellij.openapi.application.ApplicationInfo
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.ProjectManager
+import io.netty.handler.codec.http.HttpMethod
+import io.netty.handler.codec.http.QueryStringDecoder
+import org.sonarlint.intellij.issue.hotspot.SecurityHotspotOpener
+
+const val STATUS_ENDPOINT = "/sonarlint/api/status"
+const val SHOW_HOTSPOT_ENDPOINT = "/sonarlint/api/hotspots/show"
+const val PROJECT_KEY = "project"
+const val HOTSPOT_KEY = "hotspot"
+const val SERVER_URL = "server"
+
+data class Status(val ideName: String, val description: String)
+
+class RequestProcessor(private val appInfo: ApplicationInfo = ApplicationInfo.getInstance(),
+                       private val hotspotOpener: SecurityHotspotOpener = SecurityHotspotOpener()) {
+
+    fun processRequest(request: Request): Response {
+        if (request.path == STATUS_ENDPOINT && request.method == HttpMethod.GET) {
+            return getStatusData()
+        }
+        if (request.path == SHOW_HOTSPOT_ENDPOINT && request.method == HttpMethod.GET) {
+            return processOpenInIdeRequest(request)
+        }
+        return BadRequest("Invalid path or method.")
+    }
+
+    private fun getStatusData(): Response {
+        val description = ProjectManager.getInstance().openProjects.joinToString(", ") { it.name }
+        val status = Status("${appInfo.versionName} ${appInfo.fullVersion}", description)
```

Comment:
I agree about the version, that could be moved to description because it is unlikely that users will have several instances running at the same time.

I would use the same rationale for the edition name, it is also unlikely that people will run one CE and one Ultimate instance at the same time. I am OK to have it but maybe also in the description ?

Reply:
Fine by me!